blocks/library/cover-image/index.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * WordPress dependencies
+ */
+import Button from 'components/button';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import { registerBlock, query } from '../../api';
+import Editable from '../../editable';
+// TODO: Revisit when we have a common components solution
+import Dashicon from '../../../components/dashicon';
+
+const { attr, text } = query;
+
+// hack based on https://www.w3.org/TR/CSS1/#url
+// does not support escaped values
+// const URL_REGEX = /url\(\s*['"]?([^'")]*)['"]?\s*\)/i;
+
+/**
+ * Returns an attribute setter with behavior that if the target value is
+ * already the assigned attribute value, it will be set to undefined.
+ *
+ * @param  {string}   align Alignment value
+ * @return {Function}       Attribute setter
+ */
+function toggleAlignment( align ) {
+	return ( attributes, setAttributes ) => {
+		const nextAlign = attributes.align === align ? undefined : align;
+		setAttributes( { align: nextAlign } );
+	};
+}
+
+// /**
+//  * Returns the URL from a CSS url() value
+//  * e.g. url( 'https://img.youtube.com/vi/dQw4w9WgXcQ/0.jpg');
+//  *
+//  * @param {Function} func function that provides the CSS property
+//  * @return {Function}     funtion that will return a string with the url
+//  */
+// function extractUrl( func ) {
+// 	return ( node ) => {
+// 		const propValue = func( node );
+// 		const matches = URL_REGEX.match( propValue );
+// 		return matches[ 1 ];
+// 	};
+// }
+
+registerBlock( 'core/cover-image', {
+	title: wp.i18n.__( 'Cover Image' ),
+
+	icon: 'format-image',
+
+	category: 'common',
+
+	attributes: {
+		url: attr( 'section', 'data-url' ),
```

Comment:
FWIW, it's possible to drop the `url` attribute here and the url will be saved in the block opening comment.